packages/fast-usdc-contract/src/exos/advancer.ts

```diff
@@ -278,24 +313,67 @@ export const prepareAdvancerKit = (
           result: undefined,
           ctx: AdvancerVowCtx & { tmpSeat: ZCFSeat },
         ) {
-          const { poolAccount, intermediateRecipient, settlementAddress } =
-            this.state;
-          const { destination, advanceAmount, tmpSeat, ...detail } = ctx;
-          tmpSeat.exit();
-          const amount = harden({
-            denom: usdc.denom,
-            value: advanceAmount.value,
-          });
-          const transferOrSendV =
-            destination.chainId === settlementAddress.chainId
-              ? E(poolAccount).send(destination, amount)
-              : E(poolAccount).transfer(destination, amount, {
-                  forwardOpts: { intermediateRecipient },
-                });
-          return watch(transferOrSendV, this.facets.transferHandler, {
-            destination,
-            advanceAmount,
-            ...detail,
+          return asVow(async () => {
+            const { poolAccount, intermediateRecipient, settlementAddress } =
+              this.state;
+            const { tmpSeat, ...vowContext } = ctx;
+            const { destination, advanceAmount, ...detail } = vowContext;
+            tmpSeat.exit();
+            const amount = harden({
+              denom: usdc.denom,
+              value: advanceAmount.value,
+            });
+            const accountId = parseAccountIdArg(destination);
+            await null;
+
+            const chainId: CaipChainId = `${accountId.namespace}:${accountId.reference}`;
+
+            let transferOrSendV;
+            if (
+              accountId.namespace === 'cosmos' &&
+              accountId.reference === settlementAddress.chainId
+            ) {
+              // send to recipient on Agoric
+
+              const cosmosAddress = makeCosmosAccountId(
+                chainId,
+                accountId.accountAddress,
+              );
+              transferOrSendV = E(poolAccount).send(cosmosAddress, amount);
+            } else if (accountId.namespace === 'cosmos') {
+              // send via IBC
+
+              transferOrSendV = E(poolAccount).transfer(destination, amount, {
+                forwardOpts: {
+                  intermediateRecipient,
+                },
+              });
+            } else if (supportsCctp(destination)) {
+              // send USDC via CCTP
+              assert(
+                intermediateRecipient,
+                'intermediateRecipient must be set.',
+              );
+
+              await E(poolAccount).transfer(intermediateRecipient, amount);
```

Comment:
We can't await here - we need to `watch`.

A shorter path to "feat: CAIP-10 destinations" is likely coercing EUD to `accountId` and using that for destination. `.send()`, `.transfer()` should be able to handle it. 

This commit already does that, but has a little more in it that isn't 100% yet

Reply:
I've taken out CCTP changes